src/pages/Catalog/index.tsx

```diff
@@ -24,56 +24,35 @@ export default function Catalog() {
   const [modalOpened, setModalOpened] = React.useState<boolean>(false);
   const [isFiltered, setIsFiltered] = useState(true);
   const isMobile = useMediaQuery("(max-width: 36em)");
-
-  let user: {
-    categories?: string[];
-    searchQuery?: string;
-  } | null = null;
-
-  const location = useLocation();
-  if (location.state) {
-    user = location.state;
-  }
+  const navigate = useNavigate();
 
   const setSortedProducts = (products: Product[]) => {
     setProducts(products.filter((p) => !p.tags.includes("out_of_stock")).concat(products.filter((p) => p.tags.includes("out_of_stock"))));
   };
 
-  useEffect(() => {
-    if (user?.searchQuery) {
-      getProducts().then((products) => {
-        setSortedProducts(
-          products.filter((product) =>
-            `${product.name}${product.custom_id && ` - ${product.custom_id}`}`.toLowerCase().includes(user!.searchQuery!.toLowerCase()),
-          ),
-        );
-      });
-      setIsLoaded(true);
-    }
-    if (user?.categories) {
-      // fetch with first category
-      getProducts().then((products) => {
-        products = products.filter((product) => !!product.expand?.category);
-
-        // filter products with other categories
-        for (const category of user!.categories!) {
-          products = products.filter((product) => product.expand!.category!.find((productCategory) => productCategory.name === category));
-        }
+  const searchProducts = () => {
+    const search = decodeURIComponent(window.location.href.split("?search=")[1].toLowerCase());
```

Comment:
i don't think this is a good idea (could be `&search=`), use `Url` class instead maybe?